### PR TITLE
Added async calls for the server check

### LINKED ISSUE #464: Server not always loading up instantly after the systemd service is executed
#### Describe the bug
Server not always loading up instantly after the systemd service is executed and the pi changes to the online server instead.

#### To Reproduce
1. Try restarting the speaker again and again.
2. it will show a line saying server is down

#### Expected behavior
To connect to the local server as soon as available.

#### Additional context
I'll be writing an async method that switches to local server as soon as available.

### PR DESCRIPTION
**Pull request by sansyrox**
Fixes #464

#### Checklist

- [x] I have read the Contribution & Best practices Guide and my PR follows them.
- [x] My branch is up-to-date with the Upstream master branch.
- [x] The unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Test Passing

- [x] The SUSI Server must be building on the pi on bootup
- [x] The hotword detection should have a decent accuracy
- [ ] SUSI Linux shouldn't crash when switching from online to offline and vice versa (failing as of now)
- [ ] SUSI Linux should be able to boot offline when no internet connection available (failing)

### CODE REVIEW

**Review by hongquan — CHANGES_REQUESTED** [review]
1. The local server URL `https://127.0.0.1:4000` should be defined as a constant and reused.

2. Instead of

    ```py
    requests.get('/susi/chat.json?q={}&timezoneOffset={}'.format(test_params.q, test_params.timezoneOffset))
    ```
    Let's do:
    ```py
    from urllib.parse import urlencode

    LOCAL_SERVER_URL = 'https://127.0.0.1:4000'

    test_params = {
    'q': 'Hello',
    'timezoneOffset': int(time.timezone / 60)
    }
    requests.get('{}/susi/chat.json?{}'.format(LOCAL_SERVER_URL, urlencode(test_params))
    ```
    or later, if we upgrade to Python 3.6:
    ```py
    from urllib.parse import urlencode

    LOCAL_SERVER_URL = 'https://127.0.0.1:4000'

    test_params = {
    'q': 'Hello',
    'timezoneOffset': int(time.timezone / 60)
    }
    requests.get(f'{LOCAL_SERVER_URL}/susi/chat.json?{urlencode(test_params)}'
    ```

**Comment by stealthanthrax** [comment]
Sure @hongquan . I also want to test this PR first(I'll do it at the weekend) and make the changes suggested by you 🖖

**Comment by stealthanthrax** [comment]
Sorry for not updating it for a long time. @hongquan , I have updated it. Please have a look.

File: main/states/susi_state_machine.py [hunk 1]
```diff
@@ -80,6 +87,25 @@ def __init__(self, renderer=None):
         else:
             logger.warning("Susi has the wake button disabled")
             self.wake_button = None
+    
+    def server_checker(self):
+        response_one = None
+        test_params = {
+        'q': 'Hello',
```
Comment (hongquan) [inline main/states/susi_state_machine.py]: Please care the indentation (coding style)

File: main/states/susi_state_machine.py [hunk 2]
```diff
@@ -80,6 +87,25 @@ def __init__(self, renderer=None):
...
+        'timezoneOffset': int(time.timezone / 60)
+        }
+        while response_one is None:
+            try:
+                response_one = requests.get('{}/susi/chat.json?{}'
+                .format(self.server_url,urlencode(test_params)) ).result()
```
Comment (hongquan) [inline main/states/susi_state_machine.py]: Please care the whitespace (coding style)

**Review by hongquan — CHANGES_REQUESTED** [review]

**Comment by stealthanthrax** [comment]
@hongquan , updated

**Comment by hongquan** [comment]
Thanks, @stealthanthrax

**Comment by norbusan** [comment]
Hi @stealthanthrax 
this change produced some regression - does it work on Raspi? *With* this patch I get communication errors between susi server and susi linux:

With patch evey 10s sec the following warning on the server side
```
Apr 16 13:51:11 burischnitzel start.sh[14160]: 2019-04-16 13:51:11.146 WARN org.eclipse.jetty.http.HttpParser  - Illegal character 0x16 in state=START for buffer HeapByteBuffer@626b966c[p=1,l=319,c=8192,r=318]={\x16<<<\x03\x01\x01:\x01\x00\x016\x03\x03\xB2{\x8f\xBd\x80\x1a\xA9...\x02X>.\xC4egm{\xC7\xC5\x82j\x9d1>>>tent-Type: applic...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00}
Apr 16 13:51:11 burischnitzel start.sh[14160]: 2019-04-16 13:51:11.147 WARN org.eclipse.jetty.http.HttpParser  - bad HTTP parsed: 400 Illegal character 0x16 for HttpChannelOverHttp@2fe04398{r=0,c=false,a=IDLE,uri=null}
```
and on the susi linux side:
```
2019-04-16 13:52:05 DEBUG:urllib3.connectionpool: Starting new HTTPS connection (1): 127.0.0.1:4000
2019-04-16 13:52:15 DEBUG:urllib3.connectionpool: Starting new HTTPS connection (1): 127.0.0.1:4000
2019-04-16 13:52:25 DEBUG:urllib3.connectionpool: Starting new HTTPS connection (1): 127.0.0.1:4000
2019-04-16 13:52:35 DEBUG:urllib3.connectionpool: Starting new HTTPS connection (1): 127.0.0.1:4000
...
```

This is on my desktop, not a raspi, though.
Do you have an explanation for that?

**Comment by norbusan** [comment]
It seems that something in the encoding of the request is going wrong.

File: main/states/susi_state_machine.py (hunk 1 above)
Comment (norbusan) [reply main/states/susi_state_machine.py]: This has not been resolved

File: main/states/susi_state_machine.py (hunk 2 above)
Comment (norbusan) [reply main/states/susi_state_machine.py]: Same here, not changed by the last commit cb23f58

**Pull request merged**